test/Common.cpp

```diff
@@ -117,9 +117,20 @@ bool CommonOptions::parse(int argc, char const* const* argv)
 
 	po::command_line_parser cmdLineParser(argc, argv);
 	cmdLineParser.options(options);
-	po::store(cmdLineParser.run(), arguments);
+	auto parsedOptions = cmdLineParser.run();
+	po::store(parsedOptions, arguments);
 	po::notify(arguments);
 
+	for (auto const& parsedOption: parsedOptions.options)
+		if (parsedOption.position_key >= 0)
```

Comment:
Note to self: From boost docs

> Position key of this option. All options without an explicit name are sequentially numbered starting from 0. If an option has explicit name, 'position_key' is equal to -1. It is possible that both position_key and string_key is specified, in case name is implicitly added.

So, if this if condition is true, that means a non-named boost option was specified. Named options for isoltest are `editor`, `help`, `no-color`, and `test`/`t`

So if commad line option is anything other than these four strings, that option's position key is going to be non-negative.

Reply:
Acutally I think if there is any option like ``--something`` where "something" is not among these four strings, then boost will raise "unrecognized options" on its own... only if there is any option *without* a prefix ``--`` in the beginning *or* if the option comes *after* the `` -- `` delimiter, it will have ``position_key >= 0``.

And by the way: boost docs really suck IMO ;-).

Reply:
Haha... I was just trying to make a more informed review and boost docs was the first Google hit ;-)

You are right, it's not entirely clear what the expectation should be from the boost docs...

Reply:
Yeah... I was reading through it as well and in the end figured that I basically had to try and see what happens, since the docs are really far from clear and exhaustive...